mysql-test/suite/innodb/t/crypt_data_redo.test

```diff
@@ -0,0 +1,57 @@
+--echo #
+--echo # PS-7338 : Assertion failure: os0file.cc:1767:key_version_read_from_page != ENCRYPTION_KEY_VERSION_NOT_ENCRYPTED
+--echo #
+
+--source ../include/crypt_data_redo.inc
+
+--echo # record table checksums so that we can verify post crash recovery
+
+create table checksum(id INT NOT NULL AUTO_INCREMENT PRIMARY KEY, a BIGINT);
+
+--disable_query_log
+--let $i = 1
+while ($i <= 10) {
+--let $original_checksum = query_get_value(CHECKSUM TABLE tt_$i, Checksum, 1)
+--eval insert into checksum values (NULL, $original_checksum)
+--inc $i
+}
+--enable_query_log
+
+--source include/kill_and_restart_mysqld.inc
+
+--echo # verify checksums
+--let $i = 1
+while ($i <= 10) {
+--let $new_checksum = query_get_value(CHECKSUM TABLE tt_$i, Checksum, 1)
+--let $old_checksum = query_get_value(SELECT a FROM checksum WHERE id = $i, a, 1)
+--let $assert_text = Table checksum must match after crash recovery
+--let $assert_cond = $old_checksum = $new_checksum
+--source include/assert.inc
+--inc $i
+}
+
+--echo # test cleanup
```

Comment:
Most of the data being cleaned up here comes from 'inc' file. Would it be better to move those data cleanup into a separate 'inc' file as well?

Reply:
Yeah. frankly, inc file is not necessary. I just merged all .inc's by mohit